File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int max=0;
        for(int i=0;i<s.length();i++){
            HashSet<Character> set=new HashSet<>();
            int count=0;
            for(int j=i;j<s.length();j++){
                if(set.contains(s.charAt(j))){
                    break;
                }
                set.add(s.charAt(j));
                count++;
            }
            if (count > max) {
                max = count;  
            }
        }
        return max;
    }
}

/* class Solution {
    public int lengthOfLongestSubstring(String s) {
        int maxLen=0;
        HashSet<Character> set=new HashSet<>();
        int start=0;
        int end=0;
        while(end<s.length()){
            if(set.contains(s.charAt(end))){
                set.remove(s.charAt(start));
                start++;
            }else{
                set.add(s.charAt(end));
                end++;
                maxLen=maxLen<set.size()?set.size():maxLen;
            }
        }
        return maxLen;
    }
} */

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        visited=set()
        start=0
        end=0
        maxi=0
        while(end<len(s)):
            if(s[end] in visited):
                visited.remove(s[start])
                start+=1
            else:
                visited.add(s[end])
                end+=1
                if(maxi<len(visited)):
                    maxi=len(visited)
        return maxi
        